notes/benchmarks/benchmarks_kpis.md

```diff
@@ -0,0 +1,71 @@
+# Document and validate KPIs for Deku benchmarks
+
+Open [issue 574](https://github.com/marigold-dev/deku/issues/574)
+
+## Summary
+
+Define KPIs (Key Performance Indicators) to benchmark various parts of Deku (consensus, hashing...)
+
+## Motivation
+
+We want to measure Deku's ability to handle an important number of transactions. This number depends on several factors (number of validators, hardware, network...). We want to establish a protocol to make this measurements and find potential bottlenecks in the project.
+
+## Production
+
+We define the KPIs into 3 states, and by priority order.
+
+### State 1
+
+Focus on getting the number of TPS, hardware configurations, etc.
```

Comment:
I'd be more specific: can you list all the factors you expect to test, and be specific about the hardware you want to change as well? Others (infra team?) will probably want to read this as well.

Reply:
- can you list all the factors you expect to test?

I think it is what I explain below:

```
For transactions, there are two kinds of transactions we are taking into account:
- Normal transaction (A sends 10 xtz to B)
- Smart contract call (A calls a smart contract C)
```

About the hardware, currently I do not have any specific configuration that we want to test. I will ask the infra team for help about it and will update the doc.

Reply:
I think we want to be able to run benchmarks in 3 different scenarios:

1.  Locally to easily see some regressions when working on performance sensitive code
2. In CI (GitHub Actions) which should be more stable to see that we don't introduce big regressions
3. In a testnet to see that we don't slow down over time. This is most likely some cronjob that creates different transactions and then we can look at graphs from  prometheus data

1 and 2 should be written with the help of some benchmarking library
3 is a e2e test that we can run periodically

Reply:
I added your comments into the doc. Thanks.

Reply:
@ulrikstrid, my thinking is that we do not know where to write benchmarks for 1 a priori. I think we should run some kind of e2e benchmarks in CI constantly, at least until we better understand the performance characteristics of the system. If every commit is too much, we can do nightly.